### ISSUE TITLE
bug(md-input) placeholder hidden underneath input with no value

### ISSUE THREAD

**Issue by benb7760** [angular/material2#3196]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
With no value provided, the placeholder should sit in front of the input element

#### What is the current behavior?
The placeholder is hidden behind the input element. Selecting the element triggers the ordinary animation, moving the placeholder above the input.

#### What are the steps to reproduce?
http://plnkr.co/edit/VV9wDNx7FIK0gVDWsMU6?p=preview
I was annoyingly unable to reproduce on Plunker. The code in that Plunker does not work on my machine. I am using the material builds repo;
"@angular/material": "git+https://github.com/angular/material2-builds.git#8f246c0"

That exact commit is where the bug first appears for me. I have also tried against the main repo at beta.2 and the same issue occurs

-I can't see any specific CSS being used within my project- I have since removed all of my app content and this still happens with no custom CSS

#### Which versions of Angular, Material, OS, browsers are affected?
Chrome 56, Safari 10.0.3 (untested elsewhere). Material 2.0.0-beta.2 and material-builds #8f246c0

**Comment by benb7760** [comment]
Update - it appears to be resolved by reverting:
`.mat-input-placeholder-wrapper::after` `content` from '' back to '\00a0'

**Comment by crisbeto** [comment]
The issue with reverting the content back to `\00a0` is that the browser can end up rendering it in some cases (AFAIK Chrome on Linux is one of them).

**Comment by benb7760** [comment]
No selection
![image](https://cloud.githubusercontent.com/assets/497125/23106619/0eaf6376-f6e8-11e6-8773-2d0ec6f2a638.png)

Selection
![image](https://cloud.githubusercontent.com/assets/497125/23106622/178abec8-f6e8-11e6-8c90-defa430ab5c5.png)

**Comment by crisbeto** [comment]
Hmm that's weird. What OS and browser are you using?

**Comment by benb7760** [comment]
Tested in both Chrome 56, Safari 10.0.3

**Comment by crisbeto** [comment]
I couldn't reproduce it with neither of those browsers. Are you sure there isn't anything else in your app that could be messing with it?

**Comment by benb7760** [comment]
I can't see anything that could be doing it. I've stripped my app back completely so that the core view is just this repro, and removed all CSS, still happening

**Comment by lukaskopenec** [comment]
I have the same issue Chrome 56, Windows 10

**Comment by crisbeto** [comment]
We did some changes to the `.mat-input-placeholder` in beta.2 in order to prevent it from overflowing in IE 11. Still, my best guess is that there's something else influencing it, because I can't reproduce it in a Plunkr or against our development app.

**Comment by crisbeto** [comment]
Thinking about it, something worth trying could be to add this to your CSS and seeing if it helps:

```css
.mat-input-placeholder-wrapper {
  z-index: 1;
}
```

**Comment by crisbeto** [comment]
Actually, I think I found what may be causing it. Can @lukaskopenec and @benb7760 check that your index.html has a doctype?

**Comment by lukaskopenec** [comment]
@crisbeto That was it! I had a doctype but with a typo. Thanks for your help!

**Comment by benb7760** [comment]
@crisbeto Yep that has solved it here too. My DOCTYPE had some extra stuff in it which I have since removed. For reference, it looked like this;

`<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">`

and I have changed it to

`<!DOCTYPE html>`

**Comment by naiteluo** [comment]
I have the same issue. In my case, i forgot to add `<!DOCTYPE html>` in front of the html tag, after i add the correct doctype, problem solved.

**Comment by kara** [comment]
Okay, it looks like this issue was solved by fixing the DOCTYPE?  Will close then.

**Comment by xtianus79** [comment]
can i just make the notation here that quickstart... lol, which I started my projects from must have not had it in there.  just sayin!!!

**Comment by Stexxen** [comment]
I had these symptoms, but it was unrelated to the DOCTYPE.
Simply the use of a prebuilt dark theme, purple-green or pink-bluegrey will do 
and not putting the form inside a container i.e. a md-card

See http://plnkr.co/edit/fElo3SpEtDA5Fy4oLoHb?p=preview for an example

![image](https://cloud.githubusercontent.com/assets/4494301/23606686/bd2c30ba-0259-11e7-8f52-de827747e6dc.png)

Enclosing the form within a container resolved it.

**Comment by Gallopaway** [comment]
had the same problem, I didnt even have a doctype on my index page!! problem solved when i added it

**Comment by prograamer** [comment]
The same problem on Firefox 53.0.2

placeholder is shown when property of css class is disabled

.mat-input-placeholder-wrapper - overflow: hidden

**Comment by blievrouw** [comment]
Like @Stexxen, I also experience these symtoms unrelated to the doctype. Without a wrapper like md-card, the input field doesn't have the required behaviour with all themes:
(purple-green vs indigo-pink)
<img width="783" alt="screen shot 2017-06-16 at 11 53 27" src="https://user-images.githubusercontent.com/20205094/27221969-d36e7e2e-528a-11e7-8def-5ee14412ce08.png">
<img width="541" alt="screen shot 2017-06-16 at 12 06 32" src="https://user-images.githubusercontent.com/20205094/27222382-5221f574-528c-11e7-8c6f-9aad86f87642.png">

**Comment by mayankjoshi24** [comment]
Its happening with me too. Quite irritated by this bug or tweak , whatever it may be!
If someone conquered in removing this bug ,please help.

**Comment by willshowell** [comment]
@mayankjoshi24 can you reproduce and open a separate issue?

**Comment by xtianus79** [comment]
@mayankjoshi24 is this the bug wher eyou need to have this in your index.html file?

<!DOCTYPE html>
<html>

i.e. html5

**Comment by mayankjoshi24** [comment]
Below are the 2 images of my code and the result of that code in the browser.
When i reload my login page, only the Sub-domain placeholder is visible, others are visible once you click on them.

![image](https://user-images.githubusercontent.com/31066148/30026508-d35016ce-919a-11e7-9c1a-71de7660dbf7.png)

![login](https://user-images.githubusercontent.com/31066148/30026504-cedab0d6-919a-11e7-803d-62c6a35325c9.png)

**Comment by stwalter01** [comment]
I have the same problem, I am trying to clear a reactive form and simple rebuild the form, It clears the fields but anything that had a value is only an underline.  If I click in it and then click away the placeholder returns as it should.

**Comment by bertyhell** [comment]
I had the same issue unrelated to the html doctype

The problem was solved by importing the animation module.

```typescript
@NgModule({
  declarations: [
	...
  ],
  imports: [
	BrowserAnimationsModule,
	...
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```

source: https://stackoverflow.com/a/43241908/373207

**Comment by codemental** [comment]
I have the exact same situation as @stwalter01. I have a form inside a **mat-card-content** like this:

```html
<mat-card-content>
    <form [formGroup]="formGroup">
      ...
    </form>
</mat-card-content>
```

If the form shows some data and afterwards I clear it with an empty object like this:

```ts
let location: Location = new Location();

this.formGroup = new FormGroup({
      id: new FormControl(location.id),
      created: new FormControl(location.created),
      name: new FormControl(location.name, Validators.required),
...
```

the form looks like this:
![image](https://user-images.githubusercontent.com/15732974/33480942-a829c4f8-d692-11e7-9f98-ae4ccce62e30.png)
instead of this:
![image](https://user-images.githubusercontent.com/15732974/33480965-c6ed83ac-d692-11e7-95fc-1349f3e673bf.png)

If I refresh the app (CTRL-F5) the form appears correctly. This problem only appears if I navigate to it from somewhere else.

I have the _doctype_ declaration and the _BrowserAnimationsModule_ import.

OS: Linux Ubuntu 17.10
Browsers: Latest Firefox Quantum: Developer Edition and latest Chrome.

**Comment by rcanacci** [comment]
I am having the same issue and it is not the result of the missing DOCTYPE or the BrowserAnimationModule not being present.

**Issue description**
If I use one of the stock angular material themes such as:
@import '~@angular/material/prebuilt-themes/deeppurple-amber.css';
... my inputs appear as follows...

![image](https://user-images.githubusercontent.com/8365271/33863181-8c1690be-deb4-11e7-9ce9-50aabc898f22.png)

... as soon as I try to implement a custom theme, here is what happens:

![image](https://user-images.githubusercontent.com/8365271/33863224-ba86b186-deb4-11e7-9fe9-e1a19e510625.png)

... as you can see, its basically the same as above... the matInputs are there but invisible.

Clicking into them does begin the animation and properly display them (one at a time).

**Comment by maknik** [comment]
I have the same problem as @stwalter01 
After .patchValue() on react form, labels still have class ".mat-empty .mat-form-field-empty". Then, when I focus input field ".mat-empty .mat-form-field-empty" class is removed from the label. After clear all form (by [FormGroup].reset()), placeholder from previously focued field dissapear, because this field doesn't have ".mat-empty" class anymore.
I notice this only after .patchValue() on form. When I fill form manually and then do .reset(), labels are properly chcecked and ".mat-empty" class is present.

**Comment by rcanacci** [comment]
Hey, @maknik, the issue has been re-opened here:

https://github.com/angular/material2/issues/8960

**Comment by maknik** [comment]
@rcanacci but #8960 is about hidden fields not about hidden placeholder. I'm not sure this is the same problem

**Comment by rcanacci** [comment]
@maknik  - no problem, just a good heads up then that this issue is closed. I was advised to open a new one. Cheers!

**Comment by dhananshan** [comment]
easy fix could be applying the below in your global CSS file. This will overwrite the 

> mat-form-field-placeholder

```
.mat-form-field-placeholder {
    display:block !important;
}
```

**Comment by abparasar** [comment]
This may happen when your padding properties are messing with the padding of the placeholder text.

Try this if it help:

.mat-form-field-placeholder {
  padding: 0px !important;
}
